pkg/forklift-api/webhooks/validating-webhook/admitters/plan-admitter.go

```diff
@@ -0,0 +1,118 @@
+		return util.ToAdmissionResponseAllow()
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Info("Couldn't get the cluster configuration", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
+	}
+
+	err = api.SchemeBuilder.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Info("Couldn't build the scheme", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
+	}
+	err = apis.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Info("Couldn't add forklift API to the scheme", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
+	}
+
+	cl, err := client.New(config, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		log.Info("Couldn't create a cluster client", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
+	}
+
+	sourceProvider := api.Provider{}
+	err = cl.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Source.Namespace, Name: plan.Spec.Provider.Source.Name}, &sourceProvider)
+	if err != nil {
+		log.Info("Couldn't get the source provider", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
+	}
+
+	// need to query the provider
+	if sourceProvider.Type() == api.VSphere {
+		log.Info("Provider supports all storages, passing")
+		return util.ToAdmissionResponseAllow()
+	}
+
+	destinationProvider := api.Provider{}
+	err = cl.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Destination.Namespace, Name: plan.Spec.Provider.Destination.Name}, &destinationProvider)
+	if err != nil {
+		log.Info("Couldn't get the destination provider", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
+	}
+
+	if !destinationProvider.IsHost() {
+		log.Info("Remote migration supports all storages, passing")
+		return util.ToAdmissionResponseAllow()
+	}
+
+	storageClasses := v1.StorageClassList{}
+	err = cl.List(context.TODO(), &storageClasses, &client.ListOptions{})
+	if err != nil {
+		log.Info("Couldn't get the cluster storage classes", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
+	}
+
+	storageMap := api.StorageMap{}
+	err = cl.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Map.Storage.Namespace, Name: plan.Spec.Map.Storage.Name}, &storageMap)
+	if err != nil {
+		log.Info("Couldn't get the storage class", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
+	}
+	storagePairList := storageMap.Spec.Map
+	for _, storagePair := range storagePairList {
+		scName := storagePair.Destination.StorageClass
+		for _, sc := range storageClasses.Items {
+			if scName == sc.Name && sc.Provisioner == "kubernetes.io/no-provisioner" {
+				log.Info(fmt.Sprintf("storage class '%s' is static, failing", scName))
+				return &admissionv1.AdmissionResponse{
+					Allowed: false,
+					Result: &metav1.Status{
+						Code:    http.StatusBadRequest,
+						Message: fmt.Sprintf("Incorrect destination 'StorageClass' value. Static storage class is not supported"),
```

Comment:
how about adding the name of the storage class? maybe even to include the names of all the problematic storage classes

Reply:
it's logged in the api pod, but sure we can add it here as well. i think it's better to tell the specific bad used SC other than telling all of them.

Reply:
so if I define a storage mapping with two target storage classes that don't support dynamic provisioning, I'll get the error about the first, change/fix it, and then try again to get the error about the second and do the same, right? why not returning a more helpful error here like `This plan requires dynamic volume provisioning. Therefore the following destination storage classes are invalid: ...`